### PR TITLE
SD2 fixes for waypoint 0 not beeing used anymore / MySQL 8 support

### PR DESCRIPTION
**Pull request by stealthgun**
## 🍰 Pullrequest
This PR has all the SD2 scripts that had waypoint 0 in them fixed.
Also in this PR is an update to the MySQL library so it will support MySQL 8.

This PR is related to https://github.com/cmangos/mangos-tbc/pull/392

This PR will fix https://github.com/cmangos/issues/issues/2222

I've done the SD2 scripts per commit so its easier to track wich script is fixed with what commit. 
I am aware that this is resulting in a lot of commit beeing in this PR.

**A city of Light (example for how to test and the todo list)**
This PR does **not_fix**  https://github.com/cmangos/issues/issues/2221, to fix that issue you still need to insert one more row (now 77) with:

`INSERT INTO tbcmangos.script_waypoint (entry, pathId, pointid, position_x, position_y, position_z, orientation, waittime, script_id, comment) VALUES ('19685', '0', '77', '-1848.4', '5476.07', '0.856', '0', '40000', '0', 'end -fix for quest not completing');`

### Proof
- https://www.wowhead.com/quest=10211/city-of-light

### Issues
- fix for https://github.com/cmangos/issues/issues/2222

### How2Test
- .tele shattrath
- walk to Archmage Khadgar
- Accept quest 10211
- follow the npc

### Before testing
Make sure your database has been update to no longer contain waypoints 0.
You can do that with the following script:

USE tbcmangos;
ALTER TABLE script_waypoint
DROP PRIMARY KEY;

UPDATE tbcmangos.script_waypoint SET pointid = pointid+1 where entry in (SELECT entry FROM (SELECT entry FROM tbcmangos.script_waypoint WHERE pointid = 0) X);

ALTER TABLE script_waypoint
ADD PRIMARY KEY (entry,pathId,pointid);

ALTER TABLE creature_movement
DROP PRIMARY KEY;

UPDATE tbcmangos.creature_movement SET point = point+1 where id in (SELECT id FROM (SELECT id FROM tbcmangos.creature_movement WHERE point = 0) Y);

ALTER TABLE creature_movement
ADD PRIMARY KEY (id,point);

ALTER TABLE creature_movement_template
DROP PRIMARY KEY;

UPDATE tbcmangos.creature_movement_template SET point = point+1 where entry in (SELECT entry FROM (SELECT entry FROM tbcmangos.creature_movement_template WHERE point = 0) Z);

ALTER TABLE creature_movement_template
ADD PRIMARY KEY (entry,pathId,point);

### Todo / Checklist

- [x] The database needs to be updated to not use waypoint 0 anymore.
- [X] Test the script npc_defias_traitor
- [x] Test the script npc_isillien
- [x] Test the script npc_taelan_fordring
- [x] Test the script npc_deathstalker_erland
- [x] Test the script npc_kinelory
- [x] Test the script npc_professor_phizzlethorpe
- [x] Test the script npc_prospector_remtravel
- [x] Test the script npc_wizzlecranks_shredder
- [x] Test the script npc_gilthares
- [x] Test the script npc_therylune
- [x] Test the script npc_disciple_of_naralex
- [x] Test the script npc_shadowfang_prisoner
- [x] Test the script npc_feero_ironhand
- [x] Test the script npc_morokk
- [x] Test the script npc_willix_the_importer
- [x] Test the script npc_stinky_ignatz
- [x] Test the script npc_tapoke_slim_jahn
- [x] Test the script npc_ogron
- [x] Test the script npc_galen_goodward
- [x] Test the script npc_daphne_stilwell
- [x] Test the script mob_scarlet_trainee
- [x] Test the script npc_rinji
- [x] Test the script npc_oox17tn
- [x] Test the script npc_00x09hl
- [x] Test the script npc_oox22fe
- [x] Test the script npc_dorius_stonetender
- [x] Test the script npc_phalanx
- [x] Test the script npc_rocknot
- [x] Test the script npc_hurley_blackbreath
- [x] Test the script npc_arei
- [x] Test the script npc_ame01 https://github.com/cmangos/mangos-tbc/pull/393/commits/660ab6c1a765f8dcbbd451da6905d4cc21cb003f
- [x] Test the script npc_grimstone
- [x] Test the script npc_paoka_swiftmountain
- [x] Test the script npc_kanati
- [x] Test the script npc_lakota_windsong
- [x] Test the script npc_captured_arkonarin
- [x] Test the script npc_keeper_remulos
- [x] Test the script npc_kaya
- [x] Test the script npc_tirion_fordring
- [x] Test the script npc_garments_of_quests
- [x] Test the script npc_reginald_windsor
- [x] Test the script npc_muglash
- [x] Test the script npc_ruul_snowhoof
- [x] Test the script npc_torek
- [x] Test the script npc_prospector_anvilward https://github.com/cmangos/mangos-tbc/pull/393/commits/235a7c45292aabdbcbcd720509f4404af6b41cf0
- [x] Test the script npc_ranger_lilatha
- [x] Test the script npc_barnes
- [x] Test the script npc_wounded_blood_elf
- [x] Test the script npc_ancestral_wolf
- [ ] Test the script boss_nightbane
- [x] Test the script npc_anchorite_truuen
- [x] Test the script npc_magwin
- [x] Test the script npc_squire_rowe
- [x] Test the script npc_thrall_old_hillsbrad https://github.com/cmangos/mangos-tbc/pull/393/commits/eda1248b63a2c6bd2714e24506701fccb7e5c82b
- [x] Test the script npc_fhwoor
- [x] Test the script npc_kayra_longmane
- [x] Test the script npc_demolitionist_legoso
- [X] Test the script npc_nagrand_captive https://github.com/cmangos/mangos-tbc/pull/393/commits/163932754c40e55fd169550b754fd7c243c5b0cf
- [x] Test the script npc_taretha
- [x] Test the script npc_shaheen
- [X] Test the script npc_khadgars_servant https://github.com/cmangos/mangos-tbc/pull/393/commits/e4ef62fe092c4c2ad74ffd5a6a3beb81f899baf2 + https://github.com/cmangos/mangos-tbc/pull/393/commits/35e97e1b29b16fd5793b05c216987f2ceb1de13b
- [x] Test the script npc_custodian_of_time https://github.com/cmangos/mangos-tbc/pull/393/commits/7260bba91e90aa3b33fb4439d16a159a2ca96564
- [x] Test the script npc_drijya
- [x] Test the script npc_bessy
- [x] Test the script npc_captured_vanguard
- [x] Test the script npc_protectorate_demolitionist
- [x] Test the script npc_wilda
- [x] Test the script npc_akuno
- [x] Test the script npc_letoll
- [x] Test the script mob_bt_battle_fighter
- [x] Test the script npc_clintar_dw_spirit
- [x] Test the script npc_rizzle_sprysprocket
- [x] Test the script npc_dragonmaw_racer_muckjaw
- [x] Test the script npc_dragonmaw_racer_trope
- [x] Test the script npc_dragonmaw_racer_corlok
- [x] Test the script npc_dragonmaw_racer_ichman
- [x] Test the script npc_dragonmaw_racer_mulverick
- [x] Test the script npc_dragonmaw_racer_skyshatter
- [x] Test the script npc_skyguard_prisoner
- [x] Test the script npc_tanzar
- [x] Test the script npc_harkor
- [x] Test the script npc_ashli
- [x] Test the script npc_kraz
- [x] Test the script npc_amanishi_lookout
- [X] Test the script npc_harrison_jones_za https://github.com/cmangos/mangos-tbc/pull/393/commits/05e3da225ec45f8f168504cdfd5af5e720e9e604

### CODE REVIEW

File: src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/zulaman/zulamanScripts.cpp [hunk 1]
```diff
@@ -1174,16 +1174,16 @@ struct npc_tanzarAI : public ScriptedAI
 
                 switch (m_uiHelpShoutCounter)
                 {
-                    case 0:
+                    case 1:
```
Comment (cala) [inline src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/zulaman/zulamanScripts.cpp:1177]: Nice try, but those are possible values of `m_uiHelpShoutCounter`, point indexes are a bit further down the code. 😉

Comment (stealthgun) [reply src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/zulaman/zulamanScripts.cpp]: Should be fixed with the latest commit

**Comment by cala** [comment]
Are missing:
- MySQL 8 compliancy commit, pending approval by @cyberium 
- Nightbane script update pending testing

Everything else has been merged. A few required updates were spotted and fixed, but some errors car still persist.

**Comment by killerwife** [comment]
Nightbane script requires https://github.com/cmangos/issues/issues/2229 resolution.

**Comment by rnorden** [comment]
I think the aces in the boss_terokkAI are also broken by this.

**Comment by cala** [comment]
As far as I understand it, Nightbane is now fixed and we can close this?

**Pull request closed without merging**